### ISSUE TITLE
IDE0011 should support simplified using statements

### ISSUE THREAD

**Issue by torhovland** [dotnet/roslyn#82036]
**Brief description:**

IDE0011 does not support simplified using statements like this:

```csharp
using (_ = LogContext.PushProperty("Foo", true));
using (_ = LogContext.PushProperty("Bar", true));
await next(httpContext);
```

It only supports nested statements:

```csharp
using (_ = LogContext.PushProperty("Foo", true))
{
    using (_ = LogContext.PushProperty("Bar", true))
    {
        await next(httpContext);
    }
}
```

I think there should be an option to allow the simplified using statements while enforcing missing braces in if blocks, etc.

**Languages applicable:**

Applies to C#. Not sure about VB.

**Comment by CyrusNajmabadi** [comment]
@torhovland It's unclear to me what you want the expected behavior to be in the cases you listed.

**Comment by CyrusNajmabadi** [comment]
Also, what is the value of `csharp_prefer_braces` in your editorconfig?

**Comment by torhovland** [comment]
The cases I listed are with the default `csharp_prefer_braces = true`. If I change it to `when_multiline`, the rule does not complain.

Expected behavior: I would like to be able to configure the rule so that it allows simplified using statements while still enforcing braces in if blocks, etc. Honestly, I think this should be the case even for the default setting. Why? Because in the case of if blocks, the following code is problematic:

```csharp
if (condition)
    foo();
    bar();
```

There are no equivalent risks with simplified or combined using statements.

**Comment by CyrusNajmabadi** [comment]
> The cases I listed are with the default csharp_prefer_braces = true

I still don't know what behavior you're expecting with those input cases.

**Comment by torhovland** [comment]
I expect IDE0011 to not complain about any of my examples above.

**Comment by CyrusNajmabadi** [comment]
> I expect IDE0011 to not complain about any of my examples above.

I'm confused then.  

If you have:

```c#
using (_ = LogContext.PushProperty("Foo", true));
```

And you have `csharp_prefer_braces = true` then you're saying you prefer braces.  That using-statement is missing braces, so we will complain that you need to add them.

> There are no equivalent risks with simplified or combined using statements.

This is incorrect.  And your code exacctly demonstrates the problem.  It is easy for someone to accidentally add a `;` when they actually wanted the `using` to wrap the next statement.  e.g.:

```c#
        using (equals) ;
            Foo();

// instead of

        using (equals)
            Foo();
```

Forcing braces here is very intentional.  It is immediately apparent then that you have:

```c#
        using (equals) { ; }
            Foo();

// or

        using (equals)
		{
			;
		}

        Foo();
```

And thus, 'Foo' does not actually run in the context of the using block.

**Comment by CyrusNajmabadi** [comment]
> It also doesn't support combined using statements:

> `using (_ = LogContext.PushProperty("Foo", true), _ = LogContext.PushProperty("Bar", true))`

I don't know what form this is referring to.  This is not legal C# syntax.

**Comment by torhovland** [comment]
Interestingly, this example from Microsoft docs does indeed pass IDE0011. I'll drop the combined case from the issue, then.

```csharp
using (StreamReader numbersFile = File.OpenText("numbers.txt"), wordsFile = File.OpenText("words.txt"))
{
    // Process both files
}
```

**Comment by torhovland** [comment]
> And you have `csharp_prefer_braces = true` then you're saying you prefer braces.

I prefer braces, but I also prefer avoiding nested bracing when possible.

**Comment by torhovland** [comment]
Actually, I had the code wrong. This corrected code does indeed satisfy IDE0011. The issue can be closed. Thank you for reviewing.

```csharp
using (LogContext.PushProperty("Foo", true))
using (LogContext.PushProperty("Bar", true))
{
    await next(httpContext);
}
```

And this also works:

```csharp
using var numbersFile = File.OpenText("numbers.txt");
using var wordsFile = File.OpenText("words.txt");
await next(httpContext);
```

**Comment by CyrusNajmabadi** [comment]
> I prefer braces, but I also prefer avoiding nested bracing when possible.

I don't know what nested braces you're referring to.

**Comment by CyrusNajmabadi** [comment]
Closing as this seems to have been a mistake.
